// unenv/runtime/node/net/socket.mjs
import {Readable} from "../stream/readable.mjs";
import {Writable} from "../stream/writable.mjs";
import {mergeFns} from "../../_internal/utils.mjs";
const ReadableAndWritable = class {
  constructor(readable = new Readable(), writable = new Writable()) {
    Object.assign(this, readable);
    Object.assign(this, writable);
    this._destroy = mergeFns(readable._destroy, writable._destroy);
  }
};
Object.assign(ReadableAndWritable.prototype, Readable.prototype);
Object.assign(ReadableAndWritable.prototype, Writable.prototype);
export class Socket extends ReadableAndWritable {
  constructor(_options) {
    super();
    this.bufferSize = 0;
    this.bytesRead = 0;
    this.bytesWritten = 0;
    this.connecting = false;
    this.destroyed = false;
    this.localAddress = "";
    this.localPort = 0;
    this.remoteAddress = "";
    this.remoteFamily = "";
    this.remotePort = 0;
  }
  write(_buffer, _arg1, _arg2) {
    return false;
  }
  connect(_arg1, _arg2, _arg3) {
    return this;
  }
  end(_arg1, _arg2, _arg3) {
  }
  setEncoding(_encoding) {
    return this;
  }
  pause() {
    return this;
  }
  resume() {
    return this;
  }
  setTimeout(_timeout, _callback) {
    return this;
  }
  setNoDelay(_noDelay) {
    return this;
  }
  setKeepAlive(_enable, _initialDelay) {
    return this;
  }
  address() {
    return {};
  }
  unref() {
    return this;
  }
  ref() {
    return this;
  }
}
